clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -91,6 +94,18 @@ struct BuiltinTypeDeclBuilder {
       HLSLNamespace->addDecl(Record);
   }
 
+  CXXRecordDecl *finalizeForwardDeclaration() {
+    // Force the QualType to be generated for the record declaration. In most
```

Comment:
Which of the added testcases would fail without this? Otherwise, could we add a testcase that would fail without this.

Reply:
There are some existing tests that failed and motivated this fix:
```
  Clang :: AST/HLSL/ByteAddressBuffer-AST.hlsl
  Clang :: AST/HLSL/OutArgExpr.hlsl
  Clang :: AST/HLSL/pch.hlsl
  Clang :: AST/HLSL/pch_hlsl_buffer.hlsl
  Clang :: AST/HLSL/pch_with_buf.hlsl
  Clang :: CodeGenHLSL/builtins/ByteAddressBuffers-constructors.hlsl
```